bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolBridgeHandler.java

```diff
@@ -0,0 +1,343 @@
+            if (connection != null) {
+                connection.disconnect();
+                getThing().getThings().stream().forEach(thing -> {
+                    ThingHandler th = thing.getHandler();
+                    if (th instanceof ResolEmuEMThingHandler) {
+                        ((ResolEmuEMThingHandler) th).stop();
+                    }
+                });
+
+            }
+        } catch (IOException ioe) {
+            // we don't care about exceptions on disconnect in dispose
+        }
+    }
+
+    Locale getLocale() {
+        return locale;
+    }
+
+    /* adapter to react on connection state changes and handle received packets */
+    private class ResolConnectorAdapter extends ConnectionAdapter {
+        @Override
+        public void connectionStateChanged(@Nullable Connection connection) {
+            synchronized (ResolBridgeHandler.this) {
+                if (connection == null) {
+                    isConnected = false;
+                } else {
+                    ConnectionState connState = connection.getConnectionState();
+                    if (ConnectionState.CONNECTED.equals(connState)) {
+                        isConnected = true;
+                    } else if (ConnectionState.DISCONNECTED.equals(connState)
+                            || ConnectionState.INTERRUPTED.equals(connState)) {
+                        isConnected = false;
+                    }
+                    logger.info("Connection state changed to: {}", connState.toString());
+
+                    if (isConnected) {
+                        unconnectedReason = "";
+                    } else {
+                        unconnectedReason = "TCP connection failed: " + connState.toString();
+                    }
+                }
+                updateStatus();
+            }
+        }
+
+        @Override
+        public void packetReceived(@Nullable Connection connection, @Nullable Packet packet) {
+            if (connection == null || packet == null) {
+                return;
+            }
+            boolean packetHandled = false;
+            String thingType = spec.getSourceDeviceSpec(packet).getName(); // use En here
+
+            thingType = thingType.replace(" [", "-");
+            thingType = thingType.replace("]", "");
+            thingType = thingType.replace(" #", "-");
+            thingType = thingType.replace(" ", "_");
+            thingType = thingType.replace("/", "_");
+            thingType = thingType.replaceAll("[^A-Za-z0-9_-]+", "_");
+
+            /*
+             * It would be nice for the combination of MX and EM devices to filter only those with a peerAddress of
+             * 0x10, because the MX redelivers the data from the EM to the DFA.
+             * But the MX is the exception in this case and many other controllers do not redeliver data, so we keep it.
+             */
+            logger.trace("Received Data from {} (0x{}/0x{}) naming it {}",
```

Comment:
You could enclose this with an if, that the data is not calculated when running in normal logging mode.
```suggestion
            if (logger.isTraceEnabled()) {
                logger.trace("Received Data from {} (0x{}/0x{}) naming it {}",
```